common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationStringExpressions.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.catalyst.util;
+
+import com.ibm.icu.text.StringSearch;
+
+import org.apache.spark.unsafe.types.UTF8String;
+
+/**
+ * Static entry point for collation aware string expressions.
+ */
+public final class CollationStringExpressions {
```

Comment:
Probably some different name would be better, given that these are not expressions in Spark sense. E.g. UTF8StringCollationOperations?

Reply:
I think most of the code is in relation to `StringExpressions` collation awareness, rather than `UTF8String` operations per se
that said, `CollationStringExpressions` may sound a bit misleading too

maybe we could go with just `CollationSupport.java`? (for example, this will eventually cover `StringExpressions`, as well as `RegexpExpressions`, and possibly some other expressions as well that are in neither of these - like `Between`)

Reply:
and I believe calls such as: `CollationSupport.Contains.containsCollationAware(l, r, collationId)`

would look / sound descriptive enough